### ISSUE TITLE
sfdx:force:data:soql:query not respecting SFDX_MAX_QUERY_LIMIT

### ISSUE THREAD

**Issue by joeythomaschaske** [forcedotcom/cli#1579]
### Summary

I have set SFDX_MAX_QUERY_LIMIT to 500000 by executing

```bash
sfdx force:config:set maxQueryLimit=500000       
=== Set Config

 Name          Value  Success 
 ───────────── ────── ─────── 
 maxQueryLimit 500000 true 
 ```
 
 And then when I attempt to query 76809 records, only 2000 are returned and a warning is displayed
 
 ```bash
 sfdx force:data:soql:query -q "SELECT Id FROM Attachment" -r csv > ~/sfdx-temp/attachments.csv
 ›   Warning: The query result is missing 74809 records due to a 500000 record limit. Increase the number of records returned by setting the config value "maxQueryLimit" or the environment variable "SFDX_MAX_QUERY_LIMIT" to 76809 or greater than 500000.
 ```

### Steps To Reproduce:

1.  Set a query limit > 2000
1.  Query an object that will return more than 2000 records

### Expected result

It uses the specified `maxQueryLimit` and returns all the records and the warning is updated to accurately represent the current `maxQueryLimit` amount

### Actual result

It still uses the default 2000 record limit, and displays an inaccurate warning message for the current `maxQueryLimit` amount

### System Information

```json
{
        "cliVersion": "sfdx-cli/7.155.1",
        "architecture": "darwin-x64",
        "nodeVersion": "node-v16.15.1",
        "pluginVersions": [
                "@oclif/plugin-autocomplete 0.3.0 (core)",
                "@oclif/plugin-commands 1.3.0 (core)",
                "@oclif/plugin-help 3.3.1 (core)",
                "@oclif/plugin-not-found 1.2.6 (core)",
                "@oclif/plugin-plugins 1.10.11 (core)",
                "@oclif/plugin-update 1.5.0 (core)",
                "@oclif/plugin-warn-if-update-available 1.7.3 (core)",
                "@oclif/plugin-which 1.0.4 (core)",
                "@salesforce/sfdx-plugin-lwc-test 0.1.7 (core)",
                "alias 2.0.1 (core)",
                "apex 0.13.0 (core)",
                "auth 2.1.0 (core)",
                "community 2.0.0 (core)",
                "config 1.4.12 (core)",
                "custom-metadata 2.0.0 (core)",
                "data 2.0.3 (core)",
                "generator 2.0.1 (core)",
                "info 2.0.1 (core)",
                "limits 2.0.1 (core)",
                "org 1.13.2 (core)",
                "salesforce-alm 54.5.1 (core)",
                "schema 2.1.1 (core)",
                "sfdx-cli 7.155.1 (core)",
                "signups 1.1.2 (core)",
                "source 1.10.2 (core)",
                "telemetry 2.0.0 (core)",
                "templates 54.8.0 (core)",
                "trust 2.0.0 (core)",
                "user 2.0.2 (core)"
        ],
        "osVersion": "Darwin 21.5.0"
}
```

**Comment by joeythomaschaske** [comment]
Just tested that it respects values under 2000, but not over 2000

**Comment by AllenFrank** [comment]
isnt this a duplicate to https://github.com/forcedotcom/cli/issues/1543 ?

**Comment by cristiand391** [comment]
Yes, thanks @AllenFrank, the issue was solved in plugin-data@2.0.4 https://github.com/salesforcecli/plugin-data/
which will be included in the next rc.
